dev/service_now/lib/service_now_client.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require 'rest_client'
+
+# HTTP client to create a ServiceNow incident using the API
+# Credentials are not compulsory to support authentication through Apache proxy settings
+# Configuration parameters:
+# - `server`: URL for the ServiceNow server (required)
+# - `user`: ServiceNow API username
+# - `pass`: ServiceNow API password
+# - `pass_env`: Environment variable to use for the ServiceNow API password
+# - `auth_token`: ServiceNow API key
+# - `auth_token_env`: Environment variable to use for the ServiceNow API key
+# - `auth_header`: ServiceNow API key HTTP header. Defaults to x-sn-apikey.
+# - `timeout`: Connection and read timeout in seconds. Defaults to 30.
+# - `verify_ssl`: Whether or not the client should validate SSL certificates. Defaults to true.
+# - `proxy`: Proxy server URL. Defaults to no proxy.
+#
+class ServiceNowClient
+
+  UA = 'Open OnDemand ruby ServiceNow Client'
+  attr_reader :server, :auth_header, :client, :timeout, :verify_ssl
+
+  def initialize(config)
+    # FROM CONFIGURATION
+    @user = config[:user]
+    @pass = config[:pass]
+    @auth_token = config[:auth_token]
+    @auth_header = config[:auth_header] || 'x-sn-apikey'
+    @timeout = config[:timeout] || 30
+    @verify_ssl = config[:verify_ssl] || false
+    @server = config[:server] if config[:server]
+
+    raise ArgumentError, 'server is a required parameter for ServiceNow client' unless @server
+
+    # Allow to pass secrets securely through request headers
+    auth_token_header = config[:auth_token_header]
+    @auth_token = RequestStore.store[auth_token_header.to_sym] if auth_token_header
+    pass_header = config[:pass_header]
+    @pass = RequestStore.store[pass_header.to_sym] if pass_header
+
+    Rails.logger.info "User: #{@user} pwd: #{@pass}"
```

Comment:
not sure if it is safe to put in the logs the password

Reply:
(jaja) Not good - this was for debugging a new use case - deleting...

Reply:
fixed